extensions/iceberg/src/main/java/io/deephaven/iceberg/layout/IcebergBaseLayout.java

```diff
@@ -96,7 +105,7 @@ protected IcebergTableLocationKey locationKey(
 
                 parquetInstructions = builder.build();
             }
-            return new IcebergTableParquetLocationKey(fileUri, 0, partitions, parquetInstructions);
+            return new IcebergTableParquetLocationKey(fileUri, fileCount++, partitions, parquetInstructions);
```

Comment:
This is only correct if we are ensuring ordering, and that keys are never created more than once. I think that holds with the existing implementations, but the assumption should be documented here.